packages/go_router/lib/src/router.dart

```diff
@@ -526,8 +526,10 @@ class GoRouter implements RouterConfig<RouteMatchList> {
       // verified by assert() during the initialization.
       return initialLocation!;
     }
+    final Uri platformDefaultUri =
+        Uri.parse(WidgetsBinding.instance.platformDispatcher.defaultRouteName);
     final String platformDefault =
-        WidgetsBinding.instance.platformDispatcher.defaultRouteName;
```

Comment:
What if the deeplink is called when the app is already running? wasn't the path be missing in that case? I think you need to also modify the logic in _platformReportsNewRouteInformation

Reply:
We won't need to modify the logic in `_platformReportsNewRouteInformation()` because it receives the correct `RouteInformation` from [`_handlePushRouteInformation()`] where it is constructed using the `uri` parameter.


[`_handlePushRouteInformation()`]: https://github.com/flutter/flutter/blob/eba149fcc3dc6ce1668415a81be27e14a67d02c6/packages/flutter/lib/src/widgets/binding.dart#L771-L781

Reply:
ah interesting, thanks for reminding. Can you add a todo somewhere to remind us about location to uri migration? That way we won't break the code when that happens

Reply:
Added.